### PR TITLE
139 m15 add config to cmcdheadervalidator

### PR DESCRIPTION
**Pull request by adriansqualabs**

### CODE REVIEW

File: packages/cmcd-validator-library/src/inputValidator/headerValidator/headerValidator.js [hunk 1]
```diff
@@ -28,14 +35,17 @@ const headerValidator = (headerString, errors, warnings, warningFlag = true) =>
     }
 
     keysArray.split(',').forEach((keyVal) => {
-      if (isSeparetedCorrectly(keyVal, errors)) {
+      if (config?.specificKey && !config.specificKey?.includes(keyVal)) {
```
[the remark this answers is not in the document]
Comment (adriansqualabs) [reply packages/cmcd-validator-library/src/inputValidator/headerValidator/headerValidator.js]: solved TY! :)

**Review by mcamardon — APPROVED** [review]

File: packages/cmcd-validator-library/src/tests/testCases/headerConfigTestCases.js [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+      sf: 'd',
+      sid: 'b62ac932-1967-4368-8e9a-31df70ef2bc5',
+      st: 'v',
+      tb: 67,
+    },
+    rawData: 'GET /akamai/bbb_30fps/bbb_a64k/bbb_a64k_10.m4a HTTP/1.1\nAccept: */*\nAccept-Encoding: gzip, deflate, br\nAccept-Language: es-ES,es;q=0.9\nCMCD-Object: br=67,qualabs-br,d=4011,ot=a,tb=67\nCMCD-Request: bl=31700,dl=31700,mtp=10600,nor="..%2F300kbps%2Fsegment35.m4v"\nCMCD-Session: sf=d,sid="b62ac932-1967-4368-8e9a-31df70ef2bc5",st=v\nCMCD-Status: rtp=100\nConnection: keep-alive\nHost: dash.akamaized.net\nOrigin: https://reference.dashif.org\nReferer: https://reference.dashif.org/\nSec-Fetch-Dest: empty\nSec-Fetch-Mode: cors\nSec-Fetch-Site: cross-site\nUser-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36\nsec-ch-ua: "Not_A Brand";v="99", "Google Chrome";v="109", "Chromium";v="109"\nsec-ch-ua-mobile: ?0\nsec-ch-ua-platform: "Linux"\n',
+  },
+  },
+  {
+    description: 'Test: First stage checking if custom config is not persitent, adding customKey config #1',
```
Comment (JoaquinBCh) [inline packages/cmcd-validator-library/src/tests/testCases/headerConfigTestCases.js:550]: typo persistent

File: packages/cmcd-validator-library/src/tests/testCases/headerConfigTestCases.js [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+        sf: 'd',
+        sid: 'b62ac932-1967-4368-8e9a-31df70ef2bc5',
+        st: 'v',
+        rtp: 100,
+      },
+      rawData: 'GET /akamai/bbb_30fps/bbb_a64k/bbb_a64k_10.m4a HTTP/1.1\nAccept: */*\nAccept-Encoding: gzip, deflate, br\nAccept-Language: es-ES,es;q=0.9\nCMCD-Object: br=67,com.qualabs-br,com.qualabs-bs=20,d=4011,ot=a,tb=67\nCMCD-Request: bl=31700,dl=31700,mtp=10600,nor="..%2F300kbps%2Fsegment35.m4v"\nCMCD-Session: sf=d,sid="b62ac932-1967-4368-8e9a-31df70ef2bc5",st=v\nCMCD-Status: rtp=100\nConnection: keep-alive\nHost: dash.akamaized.net\nOrigin: https://reference.dashif.org\nReferer: https://reference.dashif.org/\nSec-Fetch-Dest: empty\nSec-Fetch-Mode: cors\nSec-Fetch-Site: cross-site\nUser-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36\nsec-ch-ua: "Not_A Brand";v="99", "Google Chrome";v="109", "Chromium";v="109"\nsec-ch-ua-mobile: ?0\nsec-ch-ua-platform: "Linux"\n',
+    },
+  },
+  {
+    description: 'Test: Second stage checking if custom config is not persited, adding customKey config #2',
```
Comment (JoaquinBCh) [inline packages/cmcd-validator-library/src/tests/testCases/headerConfigTestCases.js:592]: same typo here

**Review by JoaquinBCh — APPROVED** [review]

File: packages/cmcd-validator-library/src/tests/testCases/headerConfigTestCases.js (hunk 3 above)
Comment (adriansqualabs) [reply packages/cmcd-validator-library/src/tests/testCases/headerConfigTestCases.js]: TY!

**Pull request merged**